databricks/koalas/base.py

```diff
@@ -495,7 +495,7 @@ def __rsub__(self, other) -> Union["Series", "Index"]:
                 return -column_op(F.datediff)(self, F.lit(other)).astype("long")
             else:
                 raise TypeError("date subtraction can only be applied to date series.")
-        return column_op(Column.__rsub__)(self, other)
+        return column_op(lambda left, right: right - left)(self, other)
```

Comment:
FYI: `Column.__rsub__` doesn't support `pyspark.sql.column.Column` for second parameter.

```python
>>> kdf = ks.DataFrame({"A": [1, 2, 3, 4], "B": [10, 20, 30, 40]})
>>> sdf = kdf.to_spark()
>>> col1 = sdf.A
>>> col2 = sdf.B
>>> Column.__rsub__(col1, col2)
Traceback (most recent call last):
...
TypeError: Column is not iterable
```

Reply:
It does support:

```python
>>> Column.__rsub__(df.id, 1)
Column<'(1 - id)'>
```

It doesn't work in your case above because the instance is Spark column. In practice, that wouldn't happen because it will only be called when the first operand doesn't know how to handle Spark column e.g.) `1 - df.id`.

Reply:
Does it cause any exception?

Reply:
If we use `column_op(Column.__rsub__)(self, other)` as it is, it raises `TypeError: Column is not iterable` for the case below.

```python
>>> kser = ks.Series([1, 2, 3, 4])
>>> [10, 20, 30, 40] - kser
Traceback (most recent call last):
...
TypeError: Column is not iterable
```

Reply:
This case must be handled in lines 490-492. We can move back to `Column.__rsub__`.